### ISSUE TITLE
Params prop are passed to searchParams prop while using the Edge runtime

### ISSUE THREAD

**Issue by x-priyam** [vercel/next.js#62243]
### Link to the code that reproduces this issue

https://github.com/x-priyam/nextjs-edge-params-bug

### To Reproduce

1. Create a dynamic route: `/[slug]`

1. Enable the edge runtime on the page:

   ```JSX
   export const runtime = "edge";
   ```

1. Add `params` and `searchParams` props to the default exported component:

   ```JSX
   export default function Page({
    params,
    searchParams
   }:{
    params: {
      slug: string
    },
    searchParams: {}
   }) {
     // Code
   }
   ```

1. Add a way to see the content of `searchParams` (and params too optionally)

   ```JSX
   return(
     <pre>
       {JSON.stringify(searchParams, null, "\t")}
     </pre>
   )
   ```

    _Feel free to use any method you prefer here, `console.log()`, etc._

1. Run/build the project and navigate to the appropriate page: `/anyRandomString`

## Current:
You'll see that `searchParams` has all entries present in `params`

   ```JSON
   "searchParams": {
     "slug": "someRandomString"
   }
   ```

## Expected:
`searchParams` should be an empty object

   ```JSON
   "searchParams": {}
   ```

## tl;dr
`params` are prop used for dynamic routing, their values are going into `searchParams`. Can lead to unexpected behaviors.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home Single Language
Binaries:
  Node: 21.6.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.1-canary.61 // Latest available version is detected (14.1.1-canary.61).
  eslint-config-next: 14.1.0
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed), Other (Deployed)

**Comment by phishy** [comment]
I can confirm this. As soon as I changed my layout runtime to edge, my searchParams are filled with the params. Nothing in the docs about this.

**Comment by phishy** [comment]
Since this has been open since February, I will assume that the Vercel team gives no shits about edge function routing working properly.
